### PR TITLE
Iaguis/acirenderer

### LINKED ISSUE #10: use acirenderer?
https://github.com/coreos/rocket/pull/464 introduces a library for rendering ACIs from their embedded rootfs and dependencies. Following the discussion in https://github.com/appc/docker2aci/pull/6 , we should be able to split out this library and leverage it in docker2aci itself. This will probably be blocked on the library moving out of Rocket and upstream into the appc org somewhere (appc/spec? appc/tools?), since it is a general-purpose and not Rocket specific.

### PR DESCRIPTION
**Pull request by iaguis**
Once appc/spec#216 is merged, we can use the acirenderer lib to squash images.

Resolves #10

### CODE REVIEW

**Comment by iaguis** [comment]
Now that acirenderer is merged, we can merge this. Please review.

File: lib/docker2aci.go [hunk 1]
```diff
@@ -610,112 +621,96 @@ func writeManifest(outputWriter *tar.Writer, manifest schema.ImageManifest) erro
...
+					return fmt.Errorf("error writing header: %v", err)
+				}
+				if _, err := io.Copy(outputWriter, t.TarStream); err != nil {
+					return fmt.Errorf("error copying file into the tar out: %v", err)
+				}
+			}
+			return nil
+		}); err != nil {
+			return err
+		}
```
Comment (philips) [inline lib/docker2aci.go]: not a huge fan of this error handling style, could you pull the walker out into a named function?

File: lib/conversion_store.go [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+	}
+	f, err := os.Open(aci.path)
+	if err != nil {
+		return nil, fmt.Errorf("error opening aci: %s", aci.path)
+	}
+
+	return f, nil
+}
+
+func (ms *ConversionStore) ResolveKey(key string) (string, error) {
```
Comment (philips) [inline lib/conversion_store.go]: I want to rename all of these "key string" things to ID, I will do that against the spec here in a sec.

**Comment by philips** [comment]
overall, lgtm besides the nits. If those are addressed feel free to merge.

File: lib/conversion_store.go (hunk 2 above)
Comment (philips) [inline lib/conversion_store.go]: thinking about it more the key naming is fine.

File: lib/docker2aci.go (hunk 1 above)
Comment (iaguis) [inline lib/docker2aci.go]: While I was at it I did the same for the other walker

**Pull request merged**